src/actors/generic.py

```diff
@@ -0,0 +1,134 @@
+    """Provides a Generic Actor to simplify the process of setting up logs and
+    parsing input."""
+
+    # Setting up log when loaded in main omits all log setup,
+    # So a helper function/class must be created to load and start
+    # all the actors, including the main actor.
+    log = None
+    parent: ActorAddress = None
+    actor: ActorNames = None
+
+    # For looping parts
+    loop_period_seconds: float = 10
+    loop_enabled: bool = False
+
+    # For contacting other actors
+    address_book = None
+
+    def __init__(self, *args, **kwargs):
+        if not self.actor or self.actor not in ActorNames:
+            raise Exception("Please provide an actor:ActorNames value for this actor!")
+
+        # We can init a new log as long as children of this class provide 'name', but
+        # it's very hard to get these children to use the original thread's logging params.
+        if not self.log:
+            self.log = logging.getLogger(self.actor.name)
+
+        # Init the ActorTypeDispatcher
+        super(ActorTypeDispatcher, self).__init__(*args, **kwargs)
+        self.log.info(f"Initialized actor '{self.actor.name}'.")
+
+    def receiveMsg_str(self, message: str, sender: ActorAddress):
+        """Parses incoming messages containing strings."""
+        self.log.debug("Received str %s from sender %s", message, sender)
+
+    def receiveMsg_int(self, message: int, sender: ActorAddress):
+        """Parses incoming messages containing integers."""
+        self.log.debug("Received int %s from sender %s", message, sender)
+
+    def receiveMsg_dict(self, message: dict, sender: ActorAddress):
+        """
+        Parses incoming messages containing dictionaries.
+
+        Make sure you SUPER this method if you are overriding it.
+        """
+        self.log.debug("Received dict %s from sender %s", message, sender)
+
+        # If the dictionary has actor-name keys, assume it is an address book.
+        if list(message.keys())[0] in ActorNames:
+            self.address_book = message
+
+    def receiveMsg_AddressBook(self, message: AddressBook, sender: ActorAddress):
+        """
+        DO NOT OVERRIDE;
+
+        Parses incoming messages containing dictionaries.
+        """
+        # Currently this fails, the arriving object is empty. Needs some work.
+        self.log.debug("Received address book from sender %s", message, sender)
+        self.log.debug("Received Book has keys: %s", message.all().keys())
+        self.address_book = message
+
+    def receiveMsg_WakeupMessage(self, message: dict, sender: ActorAddress):
+        """
+        DO NOT OVERRIDE;
+
+        Continuously runs the code in the looping method.
+        """
+        if self.loop_enabled:
+            self.wakeupAfter(self.loop_period_seconds)
+        self.log.debug("Waking up.")
+        self.loop()
+
+    def receiveMsg_Requests(self, message: Requests, sender: ActorAddress):
+        """
+        DO NOT OVERRIDE;
```

Comment:
? why not override
i dont think the generic actor should be defining message handling so specifically, we can leave that to the specific actor to define how they want to implement that

Reply:
If we have a specific set of requests, we can have a set of abstract functions to override, which would be cleaner imo.

Currently the only ones here are START and STOP.
```py
start()
stop()
```

This way, each actor class is much smaller and looks like:

```
class AbcXyz(GenericActor):
    
   def start():
        
   def stop():

   def loop():
```

Reply:
I dont think sensors should be communicating with each other at all, lets leave that to the main actor. we only have one sensor, and not all actors are sensors (webgui.. the lcd...) those dont loop. lets make the generic class as simpple as possible and if necessary we can introduce more methdos into it

Reply:
and still, they shoudl be allowed to override the message as they see fit
our messages module is gonna expand on requests and responsess and etc so we dont have to get so specific early on